### ISSUE TITLE
Duplicates when maintaining history might cause too high memory use

### ISSUE THREAD

**Issue by FabienRoger** [MathisFederico/OpenCodeBlocks#119]
Currently, the SceneHistory saves around 50 snapshots of the current state of everything on screen, including images **if I'm not mistaken**.
If a notebook contains a two dozens of 2MB images, it could amount to 2.5GB of data in the history, which seems to high. For some users relying heavily on pictures, this might make the app usable.

This memory use could be drastically reduced if the app remembered deltas and not full snapshots.

**Comment by MathisFederico** [comment]
You are right, we should just copy 'references' of images with a hash table or something like that so we don't have duplicates.
Also remembering 'deltas' is a very good idea !
